lib/pure/strformat.nim

```diff
@@ -558,12 +558,16 @@ proc format*(value: string; specifier: string; res: var string) =
   ## sense to call this directly, but it is required to exist
   ## by the ``&`` macro.
   let spec = parseStandardFormatSpecifier(specifier)
+  var value = value
```

Comment:
No, because `alignString` is also used for floats that are already stringified to the requested "precision". For floats you don't want to deal with "precision" in `alignString`. 
IMHO alignString should just align strings and not modify the passed in string in any way.